### ISSUE TITLE
Bring Envoy from official release

### ISSUE THREAD

**Issue by jamsajones** [aws/aws-app-mesh-roadmap#10]
You should be able to use the offical release of [Envoy](https//github.com/envoyproxy/envoy).

**Comment by shubharao** [comment]
@CharlyF We now support version v1.9.1.0. We do not yet have an ETA for upstreaming the changes we made (primarily for Sigv4 authentication of Envoy), but this is on our high priority item list

**Comment by tsykora-verimatrix** [comment]
Is the v1.9.1.0 available outside us-west-2?

"Failed to enable App Mesh integration
Envoy container must use the official App Mesh Envoy image"

when I try to use:
111345817488.dkr.ecr.us-east-1.amazonaws.com/aws-appmesh-envoy:v1.9.1.0-prod

**Comment by lavignes** [comment]
Hi @tsykora-verimatrix 

We recommend that you use the latest Envoy image available, which is currently tagged as v1.11.1.1-prod: https://docs.aws.amazon.com/app-mesh/latest/userguide/envoy.html

It is available in all the App Mesh regions. Do you need to use v1.9.1.0 in particular?

**Comment by kamaldon** [comment]
I am attempting to you use the version v1.11.1.1 specified in the documentation but I am getting the same error
Image: 111345817488.dkr.ecr.us-west-2.amazonaws.com/aws-appmesh-envoy:v1.11.1.1-prod
"Failed to enable App Mesh integration
Envoy container must use the official App Mesh Envoy image"

**Comment by tsykora-verimatrix** [comment]
111345817488.dkr.ecr.us-east-1.amazonaws.com/aws-appmesh-envoy:v1.11.1.1-prod

Status reason | CannotPullContainerError: Error response from daemon: pull access denied for 111345817488.dkr.ecr.us-east-1.amazonaws.com/aws-appmesh-envoy, repository does not exist or may require 'docker login'
-- | --
1337

@lavignes doesn't work either

**Comment by bcelenza** [comment]
@tsykora-verimatrix Just to confirm:
1. Your ECS task is running in us-east-1 as well?
2. You have given the ECS task permissions to pull images from ECR? (see [these docs](https://docs.aws.amazon.com/AmazonECR/latest/userguide/ECR_on_ECS.html))


I ran a quick test to ensure that the ECR repository was open for pulling from all accounts:
```
$(aws ecr --region us-east-1 get-login --no-include-email --registry-ids 111345817488)
docker pull 111345817488.dkr.ecr.us-east-1.amazonaws.com/aws-appmesh-envoy:v1.11.1.1-prod
v1.11.1.1-prod: Pulling from aws-appmesh-envoy
72d97abdfae3: Already exists 
083db2dc1aa4: Pull complete 
0ea0eef1b868: Pull complete 
97858b908079: Pull complete 
cb999a5ecfd0: Pull complete 
753534a07b32: Pull complete 
Digest: sha256:afea680c7ea35fd886be0c8599beb8d9285c2d62732480cd19a9ecda65291c13
Status: Downloaded newer image for 111345817488.dkr.ecr.us-east-1.amazonaws.com/aws-appmesh-envoy:v1.11.1.1-prod
```

To be very sure your account can pull, you could try to run the commands above.

**Comment by bcelenza** [comment]
@kamaldon I was able to reproduce your issue on the ECS console, which is where I assume you are seeing it? I'm giving the team a heads up to update the validation for that field, sorry for the inconvenience.

**Comment by kamaldon** [comment]
@bcelenza My original attempt was trying to configure it through the "Configure via JSON" option which caused the failure. I tried inputting version 1.11.1 and I also clicked the Apply button and there are no errors, and I can even create the new revision but it doesn't actually update the docker image. The new revision still has the 1.9.0 version.

**Comment by tsykora-verimatrix** [comment]
> @tsykora-verimatrix Just to confirm:
> 
>     1. Your ECS task is running in us-east-1 as well?
> 
>     2. You have given the ECS task permissions to pull images from ECR? (see [these docs](https://docs.aws.amazon.com/AmazonECR/latest/userguide/ECR_on_ECS.html))
> 
> 
> I ran a quick test to ensure that the ECR repository was open for pulling from all accounts:
> 
> ```
> $(aws ecr --region us-east-1 get-login --no-include-email --registry-ids 111345817488)
> 
> docker pull 111345817488.dkr.ecr.us-east-1.amazonaws.com/aws-appmesh-envoy:v1.11.1.1-prod
> 
> v1.11.1.1-prod: Pulling from aws-appmesh-envoy
> 72d97abdfae3: Already exists 
> 083db2dc1aa4: Pull complete 
> 0ea0eef1b868: Pull complete 
> 97858b908079: Pull complete 
> cb999a5ecfd0: Pull complete 
> 753534a07b32: Pull complete 
> Digest: sha256:afea680c7ea35fd886be0c8599beb8d9285c2d62732480cd19a9ecda65291c13
> Status: Downloaded newer image for 111345817488.dkr.ecr.us-east-1.amazonaws.com/aws-appmesh-envoy:v1.11.1.1-prod
> ```
> 
> To be very sure your account can pull, you could try to run the commands above.

thank you, indeed my IamTaskExecRole (automatically generated by CDK) was limited to my service container only, after extending it for envoy container the Fargate service I created was able to pull the envoy container.

So Envoy is now running as side-car container with my application.

Last missing piece of the puzzle is the missing Proxy/ initContainer functionality without which I can't really pass any traffic to AppMesh service. (I'm not using AWS console, instead I use CDK).

**Comment by bcelenza** [comment]
@kamaldon Yup, I'm able to reproduce that issue (Apply still sets as 1.9.0). Working on getting this resolved. Thanks for the clarification!

**Comment by bcelenza** [comment]
@tsykora-verimatrix Looks like you've already found it, but support for proxyConfiguration in the CDK is logged as https://github.com/aws/aws-cdk/issues/3977.

**Comment by jamsajones** [comment]
We are done with the work required to allow App Mesh to work with Envoy upstream. Officially App Mesh auth via SigV4 will be available when milestone 1.12.0 completes (see: https://github.com/envoyproxy/envoy/issues/8042). Outstanding work is X-Ray related and I am closing this issue in favor of #21 and #95.
